http/oidc/src/main/java/org/wildfly/security/http/oidc/OidcRequestAuthenticator.java

```diff
@@ -416,4 +421,31 @@ private static boolean hasScope(String scopeParam, String targetScope) {
         }
         return false;
     }
+
+    private String combineAndReorderScopes(Set<String> allScopes, String paramValue) {
+        StringBuilder combinedScopes = new StringBuilder();
+        allScopes = addScope(paramValue, allScopes);
+
+        if (allScopes.contains(OIDC_SCOPE)) { //some OpenID providers require openid scope to be added in the beginning
+            allScopes.remove(OIDC_SCOPE);
+        }
+        combinedScopes.append(OIDC_SCOPE);
+        for (String scope : allScopes) {
+            if (scope != null && !scope.isEmpty()) {
+                if (combinedScopes.length() == 0) {
+                    combinedScopes.append(scope);
+                } else {
+                    combinedScopes.append(" " + scope);
+                }
+            }
+        }
+        return addOidcScopeIfNeeded(combinedScopes.toString());
+    }
+
+    private Set<String> addScope(String scopes, Set<String>allScopes) {
```

Comment:
I think the return value can be `void` here. This method updates the given set.

Reply:
Fixed